rmw/include/rmw/rmw.h

```diff
@@ -661,6 +661,35 @@ rmw_publisher_count_matched_subscriptions(
   const rmw_publisher_t * publisher,
   size_t * subscription_count);
 
+/// Retrieve the number of non local matched subscriptions to a publisher.
+/**
+ * Query the underlying middleware to determine how many non local subscriptions are
+ * matched to a given publisher.
+ *
```

Comment:
I think it would be better to define `what is local` means by here.

```suggestion
 * The definition of local means that in the same context.
 *
```